pkg/handlers/generic/lifecycle/ccm/nutanix/handler.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nutanix
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"text/template"
+
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	caaphv1 "github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/external/sigs.k8s.io/cluster-api-addon-provider-helm/api/v1alpha1"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/common/pkg/k8s/client"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/lifecycle/config"
+	lifecycleutils "github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pkg/handlers/generic/lifecycle/utils"
+	"github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pkg/handlers/options"
+)
+
+const (
+	defaultHelmReleaseName      = "nutanix-ccm"
+	defaultHelmReleaseNamespace = "kube-system"
+
+	// This is the name of the Secret on the remote cluster that should match what is defined in Helm values.
+	//nolint:gosec // Does not contain hard coded credentials.
+	defaultCredentialsSecretName = "nutanix-ccm-credentials"
+)
+
+var ErrMissingCredentials = errors.New("name of the Secret containing PC credentials must be set")
+
+type Config struct {
+	*options.GlobalOptions
+
+	defaultValuesTemplateConfigMapName string
+}
+
+func (c *Config) AddFlags(prefix string, flags *pflag.FlagSet) {
+	flags.StringVar(
+		&c.defaultValuesTemplateConfigMapName,
+		prefix+".default-values-template-configmap-name",
+		"default-nutanix-ccm-helm-values-template",
+		"default values ConfigMap name",
+	)
+}
+
+type provider struct {
+	client              ctrlclient.Client
+	config              *Config
+	helmChartInfoGetter *config.HelmChartGetter
+}
+
+func New(
+	c ctrlclient.Client,
+	cfg *Config,
+	helmChartInfoGetter *config.HelmChartGetter,
+) *provider {
+	return &provider{
+		client:              c,
+		config:              cfg,
+		helmChartInfoGetter: helmChartInfoGetter,
+	}
+}
+
+func (p *provider) Apply(
```

Comment:
Do we need to look at the strategy at all here?

Reply:
No the CCM API does not expose strategy, lets not add it until its absolutely necessary.